nx-X11/lib/include/xtrans/Xtrans.c

```diff
@@ -999,11 +1000,12 @@ TRANS(GetPeerAddr) (XtransConnInfo ciptr, int *familyp, int *addrlenp,
 
     if ((*addrp = malloc (ciptr->peeraddrlen)) == NULL)
     {
-	prmsg (1,"GetPeerAddr: malloc failed\n");
-	return -1;
+        prmsg (1,"GetPeerAddr: malloc failed\n");
+        free(addrp);
```

Comment:
free() not required here sind the malloc returned NULL.

Reply:
done

Reply:
same as above.